pkg/controller/controller.go

```diff
@@ -353,20 +398,47 @@ func (c *Controller) updateDeployment(namespace string, deploymentName string, r
 	}
 }
 
+// getMinWorkers gets the min workers based on the
+// velocity metric: messagesSentPerMinute
+func (c *Controller) getMinWorkers(
+	messagesSentPerMinute float64,
+	minWorkers int32,
+	secondsToProcessOneJob float64) int32 {
+
+	// disable this feature for WPA queues which have not specified
+	// processing time
+	if secondsToProcessOneJob == 0.0 {
+		return minWorkers
+	}
+
+	workersBasedOnMessagesSent := int32(math.Ceil((secondsToProcessOneJob * messagesSentPerMinute) / 60))
+	//klog.Infof("%v, workersBasedOnMessagesSent=%v\n", secondsToProcessOneJob, workersBasedOnMessagesSent)
+	if workersBasedOnMessagesSent > minWorkers {
+		return workersBasedOnMessagesSent
+	}
+	return minWorkers
+}
+
 // getDesiredWorkers finds the desired number of workers which are required
 // test case run: https://play.golang.org/p/_dFbbhb1J_8
 func (c *Controller) getDesiredWorkers(
 	queueMessages int32,
+	messagesSentPerMinute float64,
+	secondsToProcessOneJob float64,
 	targetMessagesPerWorker int32,
 	currentWorkers int32,
 	idleWorkers int32,
 	minWorkers int32,
 	maxWorkers int32,
-	disruptable bool) int32 {
+	minAvailableWorkers int32) int32 {
+
+	// overwrite the minimum workers needed based on
+	// messagesSentPerMinute and secondsToProcessOneJob
+	// this feature is disabled if secondsToProcessOneJob is not set or is 0.0
+	minWorkers = c.getMinWorkers(messagesSentPerMinute, minWorkers, secondsToProcessOneJob)
```

Comment:
some return paths like [here](https://github.com/practo/k8s-worker-pod-autoscaler/pull/62/files#diff-3c862eb54a8e0e161b534e0c67e5379eL378) and [here](https://github.com/practo/k8s-worker-pod-autoscaler/pull/62/files#diff-3c862eb54a8e0e161b534e0c67e5379eL382) doesn't restrict the return value from going below this calculated minWorkers.

Reply:
hanks, updated the rules for computation to consider `maxDisruptableWorkers`

all the returns from 
```go
getDesiredWorkers(...)
```
is
```
return convertDesiredReplicasWithRules(
    currentWorkers,
    currentWorkers,
    minWorkers,
    maxWorkers,
    maxDisruptableWorkers
)
```

https://github.com/practo/k8s-worker-pod-autoscaler/pull/62/commits/47987f9711e13285d32c48b6d7a681addaa98584